internal/controllers/printresults/print_results.go

```diff
@@ -82,9 +82,16 @@ func (pr *PrintResults) Print() (totalVulns int, err error) {
 		logger.LogWarnWithLevel(messages.MsgErrorTimeoutOccurs)
 	}
 
-	return pr.totalVulns, nil
+	return pr.totalVulns, pr.checkIfVulnarabilitiesHaveErrors(), nil
```

Comment:
Instead add a new boolean return, just return an error, and let parent calls check the error.